src/replay/replay_play.cpp

```diff
@@ -283,45 +381,136 @@ void ReplayPlay::readKartData(FILE *fd, char *next_line)
+                &time,
+                &x, &y, &z,
+                &rx, &ry, &rz, &rw,
+                &speed, &steer, &w1, &w2, &w3, &w4, &skidding_state,
+                &attachment, &nitro_amount, &item_amount, &item_type, &special_value,
+                &distance, &nitro, &zipper, &skidding, &red_skidding, &jumping
+                )==26)
+            {
+                btQuaternion q(rx, ry, rz, rw);
+                btVector3 xyz(x, y, z);
+                PhysicInfo pi             = {0};
+                BonusInfo bi              = {0};
+                KartReplayEvent kre       = {0};
+
+                pi.m_speed                = speed;
+                pi.m_steer                = steer;
+                pi.m_suspension_length[0] = w1;
+                pi.m_suspension_length[1] = w2;
+                pi.m_suspension_length[2] = w3;
+                pi.m_suspension_length[3] = w4;
+                pi.m_skidding_state       = skidding_state;
+                bi.m_attachment           = attachment;
+                bi.m_nitro_amount         = nitro_amount;
+                bi.m_item_amount          = item_amount;
+                bi.m_item_type            = item_type;
+                bi.m_special_value        = special_value;
+                kre.m_distance            = distance;
+                kre.m_nitro_usage         = nitro;
+                kre.m_zipper_usage        = zipper!=0;
+                kre.m_skidding_effect     = skidding;
+                kre.m_red_skidding        = red_skidding!=0;
+                kre.m_jumping             = jumping != 0;
+                m_ghost_karts[kart_num].addReplayEvent(time,
+                    btTransform(q, xyz), pi, bi, kre);
+            }
+            else
+            {
+                // Invalid record found
+                // ---------------------
+                Log::warn("Replay", "Can't read replay data line %d:", i);
+                Log::warn("Replay", "%s", s);
+                Log::warn("Replay", "Ignored.");
+            }
         }
     }   // for i
 
 }   // readKartData
+
+//-----------------------------------------------------------------------------
+/** call getReplayIdByUID and set the current replay file to the first one
+ *  with a matching UID.
+ *  \param uid The UID to match
+ */
+void ReplayPlay::setReplayFileByUID(uint64_t uid)
+{
+    m_current_replay_file = getReplayIdByUID(uid);
+} //setReplayFileByUID
+
+//-----------------------------------------------------------------------------
+/** Search among replay file and return the first index with a matching UID.
+ *  \param uid The UID to match
+ */
+unsigned int ReplayPlay::getReplayIdByUID(uint64_t uid)
+{
+    for(unsigned int i = 0; i < m_replay_file_list.size(); i++)
+    {
+        if (m_replay_file_list[i].m_replay_uid == uid)
+        {
+            return i;
+        }
+    }
+
+    return 0; //keep the compiler happy
```

Comment:
Actually, I wouldn't say it's "keep the compiler happy", not finding a searched element seems like it should be handled error handling. Maybe return -1, or assert(false), or another way to signal it was not found?